lib/storage/providers/MemoryOnlyProvider.ts

```diff
@@ -0,0 +1,144 @@
+import _ from 'underscore';
+import sizeof from 'object-sizeof';
+import utils from '../../utils';
+import type StorageProvider from './types';
+import type {Key, KeyValuePair, Value} from './types';
+
+type Store = Record<Key, Value>;
+
+// eslint-disable-next-line import/no-mutable-exports
+let store: Store = {};
+
+const setInternal = (key: Key, value: Value) => {
+    store[key] = value;
+    return Promise.resolve(value);
+};
+
+const isJestRunning = typeof jest !== 'undefined';
+const set = isJestRunning ? jest.fn(setInternal) : setInternal;
+
+const provider: StorageProvider = {
+    /**
+     * The name of the provider that can be printed to the logs
+     */
+    name: 'MemoryOnlyProvider',
+
+    /**
+     * Initializes the storage provider
+     */
+    init() {
+        // do nothing
+    },
+
+    /**
+     * Get the value of a given key or return `null` if it's not available in memory
+     */
+    getItem(key) {
+        const value = store[key];
+
+        return Promise.resolve(value === undefined ? null : value);
+    },
+
+    /**
+     * Get multiple key-value pairs for the give array of keys in a batch.
+     */
+    multiGet(keys) {
+        const getPromises = _.map(keys, (key) => new Promise((resolve) => this.getItem(key).then((value) => resolve([key, value])))) as Array<Promise<KeyValuePair>>;
+        return Promise.all(getPromises);
+    },
+
+    /**
+     * Sets the value for a given key. The only requirement is that the value should be serializable to JSON string
+     */
+    setItem(key, value) {
+        set(key, value);
+
+        return Promise.resolve();
+    },
+
+    /**
+     * Stores multiple key-value pairs in a batch
+     */
+    multiSet(pairs) {
+        const setPromises = _.map(pairs, ([key, value]) => this.setItem(key, value));
+        return new Promise((resolve) => Promise.all(setPromises).then(() => resolve()));
+    },
+
+    /**
+     * Merging an existing value with a new one
+     */
+    mergeItem(key, _changes, modifiedData) {
+        // Since Onyx already merged the existing value with the changes, we can just set the value directly
+        return this.setItem(key, modifiedData);
+    },
+
+    /**
+     * Multiple merging of existing and new values in a batch
+     * This function also removes all nested null values from an object.
+     */
+    multiMerge(pairs) {
+        _.forEach(pairs, ([key, value]) => {
+            const existingValue = store[key] as unknown as Record<string, unknown>;
+            const newValue = utils.fastMerge(existingValue, value as unknown as Record<string, unknown>) as unknown as Value;
+
+            set(key, newValue);
+        });
+
+        return Promise.resolve([]);
+    },
+
+    /**
+     * Remove given key and it's value from memory
+     */
+    removeItem(key) {
+        delete store[key];
+        return Promise.resolve();
```

Comment:
Should this resolve with `null` like `getItem()`?

Reply:
OK, great! That sounds fine as long as that's what's happening in the other providers. Also, I just checked and I can't find any uses of `.remove()` in Expensify/App and it looks like this method is only used internally.